### DISCUSSION TITLE
Reference of a field defined in a "parent" definition not found

### DISCUSSION THREAD

**Discussion by PierreR** [cuelang/cue#957] [Q&A]
I don't understand why in the following snippet I need to redefine `a` to make it works:
```
#ServicePort: {
	a: string
	name: string
}

#ServicePort: {
        a: string // error if I remove this line
	name: "\(a)"
}

port: #ServicePort & { a: "8080" }

```
https://tip.cuelang.org/play/?id=F6u1b8CNGSO#cue@export@cue

The real context is to be able to automatically generate the name of `#ServicePort` from the `protocol` and the `port`  defined in the generated kubernetes schema (_cue.mod/gen_ directory). I would do this in the same folder hierarchy in _cue.mod/usr_):
```
#ServicePort: {
	protocol: *"TCP" | #enumProtocol
	port: int32 // error if I remove this line but why ?
	name: "\(strings.ToLower(protocol))-\(port)"
}
```

Thanks for your help.

**Comment by seh** [comment]
[Introducing a value alias](https://tip.cuelang.org/play/?id=z5q5cG3OkLW#cue@export@cue) seems to work.

**Reply by PierreR** [reply]
@seh Thanks. I will a value alias then.

I would love to understand what's going on exactly.

**Reply by seh** [reply]
It is interesting—and you probably noticed—that it works as you expected [when there's only one definition involved](https://tip.cuelang.org/play/?id=AHIjXTvBcrf#cue@export@cue).

**Reply by myitcv** [reply]
From [the spec](https://tip.cuelang.org/docs/references/spec/#declarations-and-scope):

> The scope of a field, alias or let identifier declared inside a struct literal is the innermost containing block.

My understanding is therefore that the scope of `a` is restricted to the containing braces in each case.

The value alias is exactly how I would solve this too, given the use case you describe @PierreR

**Comment by cueckoo** [comment]
This discussion has been migrated to https://github.com/cue-lang/cue/issues/957.

For more details about CUE's migration to a new home, please see https://github.com/cue-lang/cue/issues/1078.
